code/modules/mob/living/simple_animal/hostile/mining_mobs/ice demon.dm

```diff
@@ -53,7 +53,9 @@
 	if(teleport_distance <= 0)
 		return ..()
 	var/list/possible_ends = list()
-	for(var/turf/T in view(teleport_distance, target.loc) - view(teleport_distance - 1, target.loc))
+	var/list/target_turfs = get_bounding_turfs(target.loc, teleport_distance, teleport_distance)
```

Comment:
wouldn't this also get turfs that aren't visible from the loc? ie a turf that's behind a wall that obscures the view

Reply:
https://secure.byond.com/docs/ref/index.html#/proc/view

Reply:
yeah but I think it's a reasonable tradeoff that the teleporting demon can follow you out of view as long as you're still it's target

Reply:
its the view of the target location which is why i wrote it the way i did.

with this scenario the demon could teleport through a wall and then not be able to shoot you.

it honestly doesn't really matter that much though and it'd be rare in the caves to have a spot where that would happen, and it wouldn't really benefit the player to set it up anyways.

Reply:
I can do a can_see check on the picked turf, let me think about it